src/main/java/de/cronn/validation_files_diff/helper/PsiElementValidationFileFinder.java

```diff
@@ -0,0 +1,131 @@
+package de.cronn.validation_files_diff.helper;
+
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleManager;
+import com.intellij.openapi.module.ModuleUtilCore;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.roots.ContentIterator;
+import com.intellij.openapi.vfs.LocalFileSystem;
+import com.intellij.openapi.vfs.VfsUtilCore;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.openapi.vfs.VirtualFileManager;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiMethod;
+import de.cronn.validation_files_diff.JoiningStrategy;
+import de.cronn.validation_files_diff.ValidationDiffProjectOptionsProvider;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+public class PsiElementValidationFileFinder {
+
+	private final Project project;
+	private final Module module;
+	private final PsiElement element;
+
+	private PsiElementValidationFileFinder(PsiElement element, Project project, Module module) {
+		this.project = project;
+		this.module = module;
+		this.element = element;
+	}
+
+	public static PsiElementValidationFileFinder of(PsiElement element) {
+		Project project = element.getProject();
+		Module currentModule = ModuleUtilCore.findModuleForFile(element.getContainingFile());
+		return new PsiElementValidationFileFinder(element, project, currentModule);
+	}
+
+	public boolean hasCorrespondingValidationFiles() {
+		return !findCorrespondingValidationFiles().isEmpty();
+	}
+
+	public List<VirtualFile> findCorrespondingValidationFiles() {
+		VirtualFile moduleRoot = getModuleRoot();
+		if (moduleRoot == null) {
+			return Collections.emptyList();
+		}
+
+		List<String> validationFilePathPrefixes = Arrays.stream(JoiningStrategy.values()).map(this::parseValidationFilePrefix).toList();
```

Comment:
With different joining strategies defined, it is possible now that a file is used multiple times, leading to multiple occurrences in the dedicated dialog. We should add some `.distinct()` here in order to mitigate this.

Also, some formatting.

```suggestion
		List<String> validationFilePathPrefixes =
				Arrays.stream(JoiningStrategy.values())
					  .map(this::parseValidationFilePrefix)
					  .distinct()
					  .toList();
```

Reply:
Good catch, c945928ff4f8d7e3a3c99170d2ca7c8db3afa0a6

Reply:
Done good catch, c945928ff4f8d7e3a3c99170d2ca7c8db3afa0a6